cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaMirrorMaker2AssemblyOperator.java

```diff
@@ -0,0 +1,265 @@
+
+                    this.maybeUpdateStatusCommon(resourceOperator, kafkaMirrorMaker2, reconciliation, kafkaMirrorMaker2Status,
+                        (mirrormaker2, status) -> new KafkaMirrorMaker2Builder(mirrormaker2).withStatus(status).build()).setHandler(statusResult -> {
+                            // If both features succeeded, createOrUpdate succeeded as well
+                            // If one or both of them failed, we prefer the reconciliation failure as the main error
+                            if (reconciliationResult.succeeded() && statusResult.succeeded()) {
+                                createOrUpdatePromise.complete();
+                            } else if (reconciliationResult.failed()) {
+                                createOrUpdatePromise.fail(reconciliationResult.cause());
+                            } else {
+                                createOrUpdatePromise.fail(statusResult.cause());
+                            }
+                        });
+                });
+        return createOrUpdatePromise.future();
+    }
+
+    private Future<ReconcileResult<ServiceAccount>> mirrorMaker2ServiceAccount(String namespace, KafkaMirrorMaker2Cluster mirrorMaker2Cluster) {
+        return serviceAccountOperations.reconcile(namespace,
+                KafkaMirrorMaker2Resources.serviceAccountName(mirrorMaker2Cluster.getCluster()),
+                mirrorMaker2Cluster.generateServiceAccount());
+    }
+
+    @Override
+    protected String getServiceName(String name) {
+        return KafkaMirrorMaker2Resources.serviceName(name);
+    }
+
+    /**
+     * Reconcile all the mirror maker 2 connectors selected by the given mirror maker 2 instance.
+     * @param reconciliation The reconciliation
+     * @param connect The mirror maker 2
+     * @return A future, failed if any of the connectors' could not be reconciled.
+     */
+    @Override
+    protected Future<Void> reconcileConnectors(Reconciliation reconciliation, KafkaMirrorMaker2 kafkaMirrorMaker2) {
+        String mirrorMaker2Name = kafkaMirrorMaker2.getMetadata().getName();
+        if (kafkaMirrorMaker2.getSpec() == null) {
+            return maybeUpdateMirrorMaker2Status(reconciliation, kafkaMirrorMaker2,
+                    new InvalidResourceException("spec property is required"));
+        }
+        List<KafkaMirrorMaker2MirrorSpec> mirrors = Optional.ofNullable(kafkaMirrorMaker2.getSpec().getMirrors())
+                .orElse(Collections.emptyList());
+        String host = getServiceName(mirrorMaker2Name);
+        KafkaConnectApi apiClient = getKafkaConnectApi();
+        return apiClient.list(host, KafkaConnectCluster.REST_API_PORT).compose(deleteMirrorMaker2ConnectorNames -> {
+
+            MIRRORMAKER2_CONNECTORS.entrySet().stream()
+                    .forEach(connectorEntry -> deleteMirrorMaker2ConnectorNames.removeAll(mirrors.stream()
```

Comment:
For me the `forEach` just makes this harder to understand, and it would be better as a regular `for` statement.

Reply:
Agreed. Updated